functionaltests/src/test/java/specs/initialPendingMigration/InitialPendingMigrationFixture.java

```diff
@@ -0,0 +1,186 @@
+                        CustomFieldKeys.CANCEL_URL_FIELD, baseRedirectUrl + URLEncoder.encode(paymentName + " Cancel", "UTF-8")),
+                transactionType, transactionState, centAmount, currencyCode);
+    }
+
+    public String createPaymentPaypal(
+            String paymentName,
+            String paymentMethod,
+            String paymentCustomType,
+            String transactionType,
+            String transactionState,
+            String centAmount,
+            String currencyCode) throws Exception {
+
+        String successUrl = baseRedirectUrl + URLEncoder.encode(paymentName + " Success", "UTF-8");
+        String errorUrl = baseRedirectUrl + URLEncoder.encode(paymentName + " Error", "UTF-8");
+        String cancelUrl = baseRedirectUrl + URLEncoder.encode(paymentName + " Cancel", "UTF-8");
+
+        return createPayment(paymentName, paymentMethod, paymentCustomType,
+                ImmutableMap.of(CustomFieldKeys.SUCCESS_URL_FIELD, successUrl,
+                        CustomFieldKeys.ERROR_URL_FIELD, errorUrl,
+                        CustomFieldKeys.CANCEL_URL_FIELD, cancelUrl),
+                transactionType, transactionState, centAmount, currencyCode);
+    }
+
+    public String createSimplePayment(String paymentName, String paymentMethod, String paymentCustomType,
+                                      String transactionType, String transactionState,
+                                      String centAmount, String currencyCode) {
+        return createPayment(paymentName, paymentMethod, paymentCustomType, emptyMap(), transactionType, transactionState, centAmount, currencyCode);
+    }
+
+    public String createKlarnaPayment(final String paymentName,
+                                      final String transactionType,
+                                      final String ip,
+                                      final String lastName,
+                                      final String birthDay,
+                                      final String telephonenumber) {
+        Cart cart = createTemplateCartKlarna(lastName);
+        Order order = createOrderForCart(cart);
+
+        Payment payment = createAndSaveKlarnaPayment(cart, order, paymentName, transactionType,
+                MoneyImpl.centAmountOf(cart.getTotalPrice()).toString(),
+                cart.getTotalPrice().getCurrency().getCurrencyCode(),
+                ip, birthDay, telephonenumber);
+
+        return payment.getId();
+
+    }
+
+    private String createPayment(String paymentName, String paymentMethod, String paymentCustomType,
+                                 Map<String, String> specificCustomFields,
+                                 String transactionType, String transactionState,
+                                 String centAmount, String currencyCode) {
+        final MonetaryAmount monetaryAmount = createMonetaryAmountFromCent(Long.valueOf(centAmount), currencyCode);
+
```

Comment:
you can use the montary amount as method parameter instead of centAmount and currency..

Reply:
no, they come from HTML page fixtures, which could be used only as a string:

https://github.com/commercetools/commercetools-payone-integration/blob/ed6bb7b927a488cddc7c7a5cfbf9e40982b93275/functionaltests/src/test/resources/specs/initialPendingMigration/InitialPendingMigration.html#L43-L44

Reply:
ok